### DISCUSSION TITLE
Tile provider suggestions

### DISCUSSION THREAD

**Discussion by themisir** [fleaflet/flutter_map#987] [General]
I want to migrate our corporate apps from google maps to this package because google maps flutter support is really bad, causes lots of issues and doesn't work perfectly with the widget animations / page transitions, ...

As far as I know, google prohibits using google maps tiles with 3rd party plugins in their terms of service. Rather than that OSM tiles (*.tile.openstreetmap.org/*) doesn't look very well especially because people are more familiar with google maps look because that's what they often see.

So, which tile providers (paid or free) would you suggest that has nice looking graphics that somewhat familiar with the popular maps / navigation apps?

---

Or if there's a way to use Google Maps tiles, I would appreciate to learn more about that.

**Comment by JaffaKetchup** [comment]
I cant check right now for you, I'm on my phone at the moment, but I previously used a Static Tiles Google Maps URL (raster). It doesn't do rotation, but there are different versions for road, hybrid, and satellite. This URL was free to use. I have no idea whether it is legal or not, but the endpoint was unprotected, no API key needed, and I seem to remember a StackOverflow answer saying that you could?
EDIT: They work, you can find the links [in this comment](https://github.com/fleaflet/flutter_map/discussions/987#discussioncomment-1133684)

PS. Just a quick note, I don't think you're looking for tile providers, you're looking for tile **servers**.

**Comment by ibrierley** [comment]
Just be wary google maps uses vector riles afaik, to which flutter_map doesn't support really atm. Mapbox is decent, but I think their terms may imply you need to use their sdk. So you could go down the mapbox sdk route. I think you may struggle to get a close google maps match up depending a bit on your use case.

**Reply by themisir** [reply]
Checked mapbox, they offer static tiles api which could be used with 3rd party libraries like leaflet. Will gonna use it probably.

**Comment by JaffaKetchup** [comment]
Ok, so I've found Google Maps Static Tiles URL as the following (they are all raster tiles):
 - Road: [https://mt.google.com/vt/lyrs=m&x={x}&y={y}&z={z}](https://mt.google.com/vt/lyrs=m&x=74212&y=45589&z=17)
 - Hybrid: [https://mt.google.com/vt/lyrs=y&x={x}&y={y}&z={z}](https://mt.google.com/vt/lyrs=y&x=74212&y=45589&z=17)
 - Satellite: [https://mt.google.com/vt/lyrs=s&x={x}&y={y}&z={z}](https://mt.google.com/vt/lyrs=s&x=74212&y=45589&z=17)
 - Other types available: see https://stackoverflow.com/a/33023651/11846040

I've linked an example tile for each of the types, looking at the exact same area so you can see the difference.

**It is important to note that these violate the [Google Maps Platform Terms of Service](https://cloud.google.com/maps-platform/terms) - see https://github.com/fleaflet/flutter_map/discussions/987#discussioncomment-3954560!**

**Reply by themisir** [reply]
Thanks for the great detailed comment :D I've previously had multiple issues with Google regarding to app publishing policy or admob, etc.. and really don't want to deal with their robot support team which they just didn't care about our cases even if we supplied proof that if this is not allowed, our competitors also doing same thing which also should not be allowed but they just didin't care.

However I'm probably going to use mapbox.

**Reply by JaffaKetchup** [reply]
Fair enough 😅. Bot support can be the most annoying thing known on planet earth!

**Reply by JaffaKetchup** [reply]
Btw, You just triplicated your comment 😅
Correction: quadruplicated.

**Reply by themisir** [reply]
> Btw, You just triplicated your comment 😅

Lol, sorry for that. 😂

My internet was lagged so I clicked Reply multiple times

**Reply by androidseb** [reply]
> so I've found Google Maps Static Tiles URL as the following

Thanks for this, this would be amazing if we could use them, but I have a doubt... I found them too searching around some StackOverflow posts, but couldn't find them in any official Google documentation.

See [section 10.1.a of their terms of services](https://developers.google.com/maps/terms-20180207#10.-license-restrictions.):
> No access to APIs or Content except through the Service. You will not access the Maps API(s) or the Content except through the Service. For example, you must not access map tiles or imagery through interfaces or channels (including undocumented Google interfaces) other than the Maps API(s). 

@JaffaKetchup do you have any official Google documentation where you found these static tiles URL, or did you find them from unofficial sources like me?

**Reply by JaffaKetchup** [reply]
@androidseb I found them through unofficial sources, but also did some of my own trial and error to get some more overlay types. See https://github.com/fleaflet/flutter_map/discussions/987#discussioncomment-1133684 from above.

**Reply by androidseb** [reply]
So I went and asked this specifically to the Google Maps support team, and I got a clear answer: using the Google Maps Static Tiles URL (e.g. `https://mt.google.com/vt/lyrs=m&x={x}&y={y}&z={z}`) is **NOT allowed**.

## My question
> **Finding a way to use map tiles URLs: am I allowed to use use mt.google.com ?**
>
> Hello,
>
> I use a third-party library to display my map to which I would like to add Google maps, so I'm looking to use a map tiles server URL for that.
>
> By searching around, I found this:
> https://mt.google.com/vt/lyrs=m&x={x}&y={y}&z={z}
>
> And I noticed map tiles can be fetched by URL like this:
> https://mt.google.com/vt/lyrs=m&x=1&y=1&z=1
>
> Although after testing locally this seems to work correctly, I'm wondering if this is an allowed usage of Google Maps servers/data. I would not be caching the data anywhere, it is fetched on demand when the user moves their map view around.
>
> I was not able to find anywhere in Google's documentation that this usage is allowed, so at this point I am unable to confirm that this is allowed hence I cannot use it.
>
> Could you please clear my doubts about this? Can this tile URL server be used for a Web App? What about for an Android app or an iOS app (without using the Google Maps native libraries)?
>
> Thank you in advance for your response!

## The response (replaced the person's name with _RESPONDENT_NAME_)
> Hello,
> 
> Thank you for contacting Google Maps Platform Support. My name is _RESPONDENT_NAME_ and I am happy to assist you with your concern.
> 
> From my understanding, you would like to know if you are allowed to use Map Tiles through [mt.google.com](http://mt.google.com/). To answer your question, accessing map tiles or satellite imagery through any mechanism besides Google Maps Platform is not allowed. No map tiles may be directly accessed. Your application's access to the tiles will be blocked if it accesses them outside the Google Maps Platform. See the [Google Maps Platform Terms of Service](https://cloud.google.com/maps-platform/terms#3-license) for more details. You can check this[ link](https://developers.google.com/maps/faq#tos_tiles) as well for FAQs regarding our Terms of Service. With that, using maps tiles is not allowed for Web App, Android and iOS app without using Google Maps Platform. 
> 
> Hope this helps. Please let me know if you have any questions and I will be happy to answer them for you. 
> 
> Best regards,
> 
> _RESPONDENT_NAME_
> Google Maps Platform Support, APAC

I hope this helps clear things up!

**Reply by JaffaKetchup** [reply]
Thanks @androidseb, this is pretty much what I thought. I hope they don't close this endpoint now, ~~not that I'm using it of course~~ 😂😬.

**Comment by mminhlequang** [comment]
@JaffaKetchup Hi, how about style (https://mapstyle.withgoogle.com/) when use `mt.google.com`?

**Reply by JaffaKetchup** [reply]
@mminhlequang Looks like that makes vector map styles instead of raster.
